### DISCUSSION TITLE
How to use the InfluxDB integration?

### DISCUSSION THREAD

**Discussion by dkossman** [tagyoureit/nodejs-poolController#979] [App and O/S Config Questions]
I've read through the documentation and perhaps have missed this, but I'm not clear on how the InfluxDB integration works.
I have influx set up and running on the standard port, created a database, created an account with access to that database, enabled the interface, added the credentials and database name to the njspc config.json.  I don't see any influx-related errors when i start up njspc (as i did when i mis-entered the credentials the first time :-)).

What are the steps to have njspc record data (temperatures, pump rpm, valve status, etc?) to influx?

influx config:
```
      "influxDB": {
        "name": "InfluxDB",
        "type": "influx", 
        "enabled": true,
        "fileName": "influxDB.json",
        "options": {
          "version": 1,
          "protocol": "http",
          "host": "localhost",
          "port": 3000,
          "username": "pool",
          "password": "**********",
          "database": "pool",
          "retentionPolicy": "autogen",
          "tags": [
            {
              "name": "sourceIP",
              "value": "@bind=webApp.ip();"
            },
            {
              "name": "sourceApp",
              "value": "njspc"
            }
          ]
        },
        "uuid": "71e77885-bd81-439e-8f1a-ee7541fef3e9"
      },
```

my existing pool server code was logging temperature measurements to influx in this database, but I don't see anything coming from njspc:

```
pi@njspcPi:~ $ influx
Connected to http://localhost:8086 version 1.8.10
InfluxDB shell version: 1.8.10
> use pool
Using database pool
> show measurements
name: measurements
name
----
temperature
> show users
user admin
---- -----
pool true
> show series
key
---
temperature,host=njspcPi
>
```

```
> select * from temperature order by desc limit 1
name: temperature
time                air  host    pool return spa
----                ---  ----    ---- ------ ---
1718908797712162690 78.7 njspcPi 79.6 83.8   81.5
> 
```

see attached for how i am currently set up...
![njspcConfig](https://github.com/tagyoureit/nodejs-poolController/assets/3591136/c5a9cf88-62ca-4995-91f6-d228f873a8db)

**Comment by tagyoureit** [comment]
Your pool needs to be on to log values.  And you can set your logging level to silly or debug and influx related messages should appear.

**Comment by dkossman** [comment]
thanks.  Assuming i set the log level to debug, will it log to influx during the scheduled on times?

**Reply by tagyoureit** [reply]
This only changes the logging level for njsPC, not influx.  You would need to do that separately if you think it would help.  But it looks like nothing is being sent there so I doubt you'd see any relevant logs there either way.

**Reply by dkossman** [reply]
I see.  I was hoping that it would log equipment status (pump speed, heater on/off, valve on/off, intake and other temperatures, etc).  I’ll take a look at the code in a couple of weeks when we get back from our hiking adventure in Scotland.

**Reply by tagyoureit** [reply]
I thought you were just talking about the logging aspect, which I read as being separate from writing to the database.  It will store all the equipment changes in Influx _when it is working_.  You can see the `binding` that we use for [InfluxDB](https://github.com/tagyoureit/nodejs-poolController/blob/master/web/bindings/influxDB.json).

**Reply by dkossman** [reply]
Sorry for the confusion- thanks for clarifying- the bindings look comprehensive!  I’ll look at this again in a couple weeks.
